pkgs/misc/vim-plugins/overrides.nix

```diff
@@ -213,6 +213,29 @@ self: super: {
     meta.maintainers = with stdenv.lib.maintainers; [ minijackson ];
   };
 
+  LeaderF = super.LeaderF.overrideAttrs(old: {
+    buildInputs = [
+      python3
+    ];
+    buildPhase = ''
+      patchShebangs .
+      export PY3=ON
+      ./install.sh
+    '';
+    # HACK: Installing this plugin triggers Vim/Neovim to reload along with
+    # this plugin, which triggers autoload/lfMru.vim to load, which in turn
+    # attempts to read the g:Lf_CacheDirectory setting. That setting is set to
+    # $HOME, which during the build points to the non-existent
+    # /homeless-shelter. lfMru.vim tries to create the directory but lacks
+    # sufficient permissions, so this code forces g:Lf_CacheDirectory to point
+    # to $TMPDIR only when $HOME == /homeless-shelter, i.e., only during builds.
```

Comment:
I don't quite understand this. When and why is `$HOME` accessed?

Reply:
@timokau, `$HOME` is accessed here, when trying to get the `g:Lf_CacheDirectory` setting in vim: https://github.com/Yggdroot/LeaderF/blob/v1.21/autoload/lfMru.vim#L10. The default for that setting is `$HOME`, which is set to `/homeless-shelter` during the build.

A few lines later, the script attempts to create the directory pointed to by `g:Lf_CacheDirectory` if it doesn't already exist: https://github.com/Yggdroot/LeaderF/blob/v1.21/autoload/lfMru.vim#L13. This action sometimes fails during the build because the user often doesn't have permission to create `/homeless-shelter`. I assume that it would work if running as root, but we need to account for non-root users.

Besides, in this case, there isn't really a need to ensure that the cache directory exists. The error occurs when attempting to build `neovim` (and would probably affect `vim` builds, I assume, but I haven't checked) and the plugin is installed. The error _doesn't_ occur when trying to build this plugin. That's because the the invalid access occurs in the autoload script for the plugin. The plugin is simply being loaded during the course of opening `neovim` while `neovim` is being built, which triggers the autoload script, which in turn triggers the mkdir that results in a permission error (when building as a non-privileged user). Therefore, we don't actually need the cache dir because the error is being triggered when the plugin isn't really being used for anything; it serves no purpose in this context.

Reply:
Ah, that sounds like the remote manifest generation is the issue. Neovim is launched with a minimal vimrc (that just includes the plugins) during build to generate the remote plugin manifest.

If that is the case, other plugins are likely to have similar issues. Maybe it would be better to create a home directory for the manifest generation. Could you try adding `export HOME="$(mktemp  -d)"` before the manifest generation in `pkgs/applications/editors/neovim/wrapper.nix`? Please also add a short comment that explains why it is necessary.

Reply:
Just fyi, the `export HOME` fix was used in the following PR which appears to be close to being merged:
https://github.com/NixOS/nixpkgs/pull/72506/files#diff-a394ac424cb08e8ae710c7275ae578ff

Reply:
@FelschR, correct 🙂. I will clean up this PR and ask for this to be merged once https://github.com/NixOS/nixpkgs/pull/72506 gets merged.